frontend/src/lib/mocks/mocks.ts

```diff
@@ -62,3 +67,35 @@ export function mockEndpoints(
     ...overrides,
   }
 }
+
+export class MockMetricsManager implements MetricsManager {
+  private sessionInfo
+
+  public constructor(sessionInfo?: SessionInfo) {
+    this.sessionInfo = sessionInfo
+  }
+
+  enqueue(evName: string, evData: Record<string, any>): void {}
+  incrementDeltaCounter(deltaType: string): void {}
+  getAndResetDeltaCounter(): DeltaCounter {
+    return {}
+  }
+  clearDeltaCounter(): void {
+    throw new Error("Method not implemented.")
+  }
+  incrementCustomComponentCounter(customInstanceName: string): void {}
+  getAndResetCustomComponentCounter(): CustomComponentCounter {
+    return {}
+  }
+  clearCustomComponentCounter(): void {}
+}
+
+/** Return a mock MetricsManager implementation */
+export function mockMetricManager(sessionInfo?: SessionInfo) {
+  const mm = new MockMetricsManager(sessionInfo)
+  // @ts-expect-error
+  mm.track = jest.fn()
+  // @ts-expect-error
+  mm.identify = jest.fn()
```

Comment:
Why are these added? Are there tests relying on them? (If so, the tests are probably relying on implementation details of `SegementMetricsManager`, which they shouldn't be doing!)